### DISCUSSION TITLE
AppCenter error executing test solution with Magick.NET

### DISCUSSION THREAD

**Discussion by b4ssk3y** [dlemstra/Magick.NET#1321] [Help]
Hello, 
 I recently integrated Magick.NET-Q16-x64 in my test project to leverage the image recognition capabilities for image-based tests. 
When executing the tests locally on my machine, everything works fine, but as soon as I try to execute the same tests remotely on AppCenter, I get this error on test startup:

```
Error: Failed to prepare UI Test artifacts using command "test-cloud.exe prepare" with error message: 
Test chunking failed. Please try to delete dlls which are not necessary from the path provided in the --build-dir argument. If this error happens again, please contact support.
```

For what I read in some threads this may depend by the fact that the build directory contains more DLLs than expected... and this started to happen when I added the Nuget dependency to Magick.NET-Any-Q16
 
In VS I noticed also that the Magick.NET-Q16-x64 dll is also linked as external object in the solution:
![image](https://user-images.githubusercontent.com/94126337/209160959-45e2381b-87be-483e-8deb-bbf7e598e62b.png)
which happens only with Magick.NET NuGet pakage.

Did anyone experience the same issue? Any suggestion on how to overcome the problem ? 
Any help is much appreciated :) 

Thanks

**Comment by dlemstra** [comment]
I don't have any experience with AppCenter but I suspect that you will need to remove the Magick.Native* files that are not used. But it might be better to switch to the Magick.NET-x64-Q16 version instead.

**Reply by b4ssk3y** [reply]
Hello, 
Magick.NET-Q16-x64 is the version I am using already, as stated above.
I tried to remove the *-native* dll by setting the Exclude and Remove entries in csproj file, but surprisingly I still find that DLL in the bin/* output directory; any clue why ?

A workaround (working also on AppCenter) is to reference into the project directly the 2 necessary dlls: Magick.NET-Core and Magick.NET-Q16-x64.

Any suggestion ?

Thanks in advance

**Reply by dlemstra** [reply]
Which Magick files are in your `bin` folder? And you said that you added a reference to ` Magick.NET-Any-Q16` or is that a typo?

**Reply by b4ssk3y** [reply]
Hi, in bin/* folder I see Magick.NET-Q16.x64.dll, Magick.NET.Core.dll and Magick.Native-Q16-x64.dll, despite in my .csproj I have:

`<ItemGroup>
	  <Content Remove="C:\path\to\.nuget\packages\magick.net-q16-x64\12.2.2\build\netstandard20\..\..\runtimes\win-x64\native\Magick.Native-Q16-x64.dll" />
</ItemGroup>`

Visual Studio is version 17.4.3
(Magick.NET-Q16-AnyCpu was a typo in the second sentence of the original message).

Thanks

**Reply by dlemstra** [reply]
The `Magick.Native-Q16-x64.dll` file needs to be in your `bin/*` because this library won't work with this file.

**Reply by b4ssk3y** [reply]
Hello, maybe I misunderstood something, but:
- in your first reply you said: "you will need to remove the Magick.Native* files that are not used".
- in your last reply, conversely, you say: "The Magick.Native-Q16-x64.dll file needs to be in your bin/* because this library won't work with this file."

Can you please clarify this point better? 
Thanks

**Reply by dlemstra** [reply]
You will need to remove `Magick.Native*` files that are not used. The only one that you need is `Magick.Native-Q16-x64.dll` and when there are other `Native` files in your output folder they should be removed.
